### PR TITLE
Add within parameter to point lookup.

### LINKED ISSUE #363: Add lookup of areas near a point
Our `/point` endpoint (areas covering a point) could also be described as "all the areas within 0m of a point", so extending that with a parameter to look up areas within Nm of a point could be useful. Or it could be added to the `/nearest` endpoint.

### PR DESCRIPTION
**Pull request by dracos**
Fixes #363

### CODE REVIEW

File: mapit/views/areas.py [hunk 1]
```diff
@@ -394,8 +394,15 @@ def areas_by_point(request, srid, x, y, bb=False, format=''):
 
     q = query_args(request, format)
 
+    try:
+        within = float(request.GET.get('within', 0))
+    except ValueError:
+        raise ViewException(format, _('Bad within specified'), 400)
```
Comment (davea) [inline mapit/views/areas.py:400]: This error is a bit terse and hard to parse, something like `'Bad "within" parameter specified'` would be clearer.

File: mapit/views/areas.py [hunk 2]
```diff
@@ -394,8 +394,15 @@ def areas_by_point(request, srid, x, y, bb=False, format=''):
 
     q = query_args(request, format)
 
+    try:
+        within = float(request.GET.get('within', 0))
```
Comment (davea) [inline mapit/views/areas.py:398]: We should validate that this is >= zero, as negative values appear to cause an `InternalError: Tolerance cannot be less than zero` SQL error. Might be an idea to clamp it to an upper bound too, I'm not sure if a really high figure could otherwise be used to get every single area and if that would be a problem?

**Review by davea — CHANGES_REQUESTED** [review]
A couple of little things to make it easier and more robust is all.

File: mapit/views/areas.py (hunk 2 above)
Comment (dracos) [reply mapit/views/areas.py:398]: I'll make it a config variable so we can adjust as necessary.

File: mapit/templates/mapit/api/point.html [hunk 3]
```diff
@@ -63,6 +63,11 @@ <h3>{% blocktrans trimmed %}
                                 <i>country</i>, to restrict results to areas with particular country codes (multiple country codes separated by commas).
                             {% endblocktrans %}
                         </li>
+                        <li>
+                            {% blocktrans trimmed %}
+                                <i>within</i>, to show results within the specified distance of the point, not just the point itself (not <kbd>/box</kbd> version).
```
[the remark this answers is not in the document]
Comment (dracos) [reply mapit/templates/mapit/api/point.html]: Used a context processor and assumed 4326 or not, which the rest of the code also does.

File: mapit/views/areas.py [hunk 4]
```diff
@@ -394,10 +394,13 @@ def areas_by_point(request, srid, x, y, bb=False, format=''):
 
     q = query_args(request, format)
 
+    max_within = getattr(settings, 'MAPIT_WITHIN_MAXIMUM', 0)
     try:
         within = float(request.GET.get('within', 0))
+        if within < 0 or within > max_within:
```
Comment (davea) [inline mapit/views/areas.py:400]: This will unfortunately break the  feature entirely if `within` has a value > 0 and `MAPIT_WITHIN_MAXIMUM` is undefined (because `max_within` defaults to zero).

File: mapit/views/areas.py [hunk 5]
```diff
@@ -394,10 +394,13 @@ def areas_by_point(request, srid, x, y, bb=False, format=''):
 
     q = query_args(request, format)
 
+    max_within = getattr(settings, 'MAPIT_WITHIN_MAXIMUM', 0)
```
Comment (davea) [inline mapit/views/areas.py:397]: Would be good to document this in `general.yml-example`.

**Review by davea — CHANGES_REQUESTED** [review]
A small bit of finessing required but almost there.

File: mapit/views/areas.py (hunk 4 above)
Comment (dracos) [reply mapit/views/areas.py:400]: Yes, I don't think it should work unless you specify a maximum (for the reason you stated, that it would be possible to overload by requesting all areas with a huge distance). So it's a way of switching the feature off too.

File: mapit/views/areas.py (hunk 5 above)
Comment (dracos) [reply mapit/views/areas.py:397]: I was going to add/document it in mapit.mysociety.org's conf/general.yml and pull it into its settings there, iyswim, but yes, might as well be consistent here too. Shouldn't really have any conf/general.yml stuff in this project.

**Review by davea — APPROVED** [review]
Woo!
